### PR TITLE
Refactor Publish Server

### PR DESCRIPTION
**Pull request by hamstu**
## Description

Just moving some things around and out of the server code to clean things up. Pre for some work to come. No functional changes. (Only real change is removing Intercom as per the linked Slack thread.)

### CODE REVIEW

File: packages/server/index.html [hunk 1]
```diff
@@ -201,7 +201,6 @@
   {{{fullStoryScript}}}
   {{{notificationScript}}}
   {{{showModalScript}}}
-  {{{intercomScript}}}
```
Comment (hamstu) [inline packages/server/index.html:204]: Removing Intercom (See: https://buffer.slack.com/archives/C151RRDL1/p1587491560188200?thread_ts=1587376395.103700&cid=C151RRDL1)

Comment (albennett) [reply packages/server/index.html:204]: I think we reference intercom in a few different packages. Will removing it here cause any errors?

Comment (hamstu) [reply packages/server/index.html:204]: Good one! I'll clean those up too. 👍

File: packages/server/lib/embeds/bufferData.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  }
+  if (typeof profiles !== 'undefined') {
+    bufferData.profiles = profiles;
+  }
+
+  return `
+<script>
+  try {
+    window.bufferData = ${JSON.stringify(bufferData)};
+  } catch(e) {}
```
Comment (albennett) [inline packages/server/lib/embeds/bufferData.js:19]: Do you think we should log the error for debugging purpose?

Comment (hamstu) [reply packages/server/lib/embeds/bufferData.js:19]: Yeah, I can add that with the new Bufflogger from Eric! (This was just copied over from the old code, but I like the suggestion.)

File: packages/server/index.html (hunk 1 above)
Comment (hamstu) [reply packages/server/index.html:204]: @albennett Done! ✅

File: packages/server/lib/embeds/bufferData.js (hunk 2 above)
Comment (albennett) [reply packages/server/lib/embeds/bufferData.js:19]: Awesome 🙌

**Pull request merged**